app/routes/participate/survey/consent.js

```diff
@@ -0,0 +1,19 @@
+import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+import ENV from 'isp/config/environment';
+
+
+export default Ember.Route.extend(AuthenticatedRouteMixin, {
+    model() {
+        var session = this.modelFor('participate.survey');
+        // If participant has previously visited the survey (which requires granting consent),
+        // go directly to survey and skip consent page
+        if (ENV.featureFlags.continueSession) {
+            var frameIndex = session.get('frameIndex');
+            if (frameIndex && frameIndex >= 0) {
+                this.transitionTo('participate.survey');
```

Comment:
This is a reasonable assumption, but do we *need* to rely on a side-effect? ("there is other data so consent must have happened too, probably")

Is there a reason that we don't directly check the consent field itself? The general principle is to check the actual property you want to know about instead of a side effect, when possible.